[Docs SIG] - Add plugin docs and plugin site as SIG projects + Hacktoberfest featured project (#2514)
jenkins-infra/jenkins.io: content/sigs/docs/index.adoc

=== Ongoing projects

This sections lists some of the projects being done under umbrella of this SIG.
See the link:https://docs.google.com/document/d/1uNNo0QJKPHnNp8PGr_jLI8p3K_94ZYD-M0evZOEZ93c/edit?usp=sharing[SIG meeting notes] for more information about ongoing projects.

==== Plugin site integration with GitHub

link:https://plugins.jenkins.io/[Jenkins Plugin Site] currently uses Jenkins update center and link:https://wiki.jenkins.io/[Jenkins Wiki] as a main source of information to be displayed on the website.
We have recently added support of plugin documentation from GitHub (link:https://groups.google.com/forum/#!topic/jenkinsci-dev/VSdfVMDIW-A[announcement]), 

> but there is still a lot of improvement areas: visualizing changelogs, pulling plugin tags and maintainers list from GitHub, etc.

Reviewer: ```suggestion
but there are still many improvement areas: visualizing changelogs, pulling plugin tags and maintainers list from GitHub, etc.
```

This project is tracked in the jira:WEBSITE-637[] EPIC.

Useful links:

* link:https://github.com/jenkins-infra/plugin-site[Plugin Site Frontend]
* link:https://github.com/jenkins-infra/plugin-site-api[Plugin Site Backend] (exposes REST API for the frontend)
* link:https://github.com/jenkins-infra/update-center2[Jenkins Update Center] (serves plugin metadata)
* link:https://issues.jenkins-ci.org/issues/?jql=project%20%3D%20WEBSITE%20AND%20component%20%3D%20plugin-site%20AND%20labels%20%3D%20newbie-friendly%20and%20status%20in%20(Open%2C%20Reopened%2C%20%22To%20Do%22)[Newbie-friendly issues for the plugin site]
* link:https://issues.jenkins-ci.org/browse/JENKINS-59467[Template issue for plugin documentation migration to GitHub] (newbie-friendly)
* link:https://wiki.jenkins.io/display/JENKINS/Hosting+Plugins#HostingPlugins-Hostingthepluginpage[Wiki: Hosting the plugin page]

==== Plugin documentation on GitHub

> Currently the SIG is working on migrating plugin documentation from https://wiki.jenkins.io/ to GitHub.

Reviewer: ```suggestion
Currently the SIG is migrating plugin documentation from https://wiki.jenkins.io/ to GitHub.
```

> By supporting such source in plugin site we provide good user experience to Jenkins users who look for documentation. 

Reviewer: ```suggestion
Documentation in the plugin GitHub repository provides a good user experience for Jenkins users seeking documentation. 
```
